java.lang.RuntimeException: Package manager has died

Hi Kiall,

no idea whether this is helpful, search for "AndroidRuntime", there are some fatal error.

[logcat - 48.zip](https://github.com/kiall/android-tvheadend/files/560453/logcat.-.48.zip)

The actual crash part of this:

	--------- beginning of crash

	10-29 00:26:03.213  3187  6321 E AndroidRuntime: FATAL EXCEPTION: Thread-1132
	10-29 00:26:03.213  3187  6321 E AndroidRuntime: Process: ie.macinnes.tvheadend, PID: 3187
	10-29 00:26:03.213  3187  6321 E AndroidRuntime: java.lang.RuntimeException: Package manager has died
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:134)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.tvheadend.MiscUtils.getAppVersionName(MiscUtils.java:32)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.tvheadend.sync.EpgSyncService.initHtspConnection(EpgSyncService.java:138)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.tvheadend.sync.EpgSyncService.openConnection(EpgSyncService.java:117)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.tvheadend.sync.EpgSyncService$2.onStateChange(EpgSyncService.java:162)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.htsp.Connection.setState(Connection.java:330)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.htsp.Connection.open(Connection.java:188)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at ie.macinnes.htsp.Connection.run(Connection.java:103)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:818)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime: Caused by: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at android.os.BinderProxy.transactNative(Native Method)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at android.os.BinderProxy.transact(Binder.java:503)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at android.content.pm.IPackageManager$Stub$Proxy.getPackageInfo(IPackageManager.java:2216)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        at android.app.ApplicationPackageManager.getPackageInfo(ApplicationPackageManager.java:129)
	10-29 00:26:03.213  3187  6321 E AndroidRuntime:        ... 8 more

--- a/app/src/main/java/ie/macinnes/tvheadend/MiscUtils.java
+++ b/app/src/main/java/ie/macinnes/tvheadend/MiscUtils.java
@@ -26,16 +26,6 @@ import java.util.Map;
 
 
 public class MiscUtils {
-    public static String getAppVersionName(Context context) {
-        try {
-            String packageName = context.getPackageName();
-            PackageInfo info = context.getPackageManager().getPackageInfo(packageName, 0);
-            return info.versionName;
-        } catch (PackageManager.NameNotFoundException e) {
-            return "?";
-        }
-    }
-
     public static Map<String, String> createBasicAuthHeader(String username, String password) {
         Map<String, String> headerMap = new HashMap<String, String>();
 

--- a/app/src/main/java/ie/macinnes/tvheadend/account/AuthenticatorActivity.java
+++ b/app/src/main/java/ie/macinnes/tvheadend/account/AuthenticatorActivity.java
@@ -32,6 +32,7 @@ import java.util.List;
 
 import ie.macinnes.htsp.Connection;
 import ie.macinnes.htsp.ConnectionListener;
+import ie.macinnes.tvheadend.BuildConfig;
 import ie.macinnes.tvheadend.Constants;
 import ie.macinnes.tvheadend.MiscUtils;
 import ie.macinnes.tvheadend.R;
@@ -351,9 +352,7 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity {
             final String accountHttpPort = args.getString(Constants.KEY_HTTP_PORT);
             final String accountHttpPath = args.getString(Constants.KEY_HTTP_PATH);
 
-            final String versionName = MiscUtils.getAppVersionName(getActivity().getBaseContext());
-
-            mConnection = new Connection(accountHostname, Integer.parseInt(accountHtspPort), accountName, accountPassword, "android-tvheadend (auth)", versionName);
+            mConnection = new Connection(accountHostname, Integer.parseInt(accountHtspPort), accountName, accountPassword, "android-tvheadend (auth)", BuildConfig.VERSION_NAME);
 
             final ConnectionListener connectionListener = new ConnectionListener() {
                 @Override

--- a/app/src/main/java/ie/macinnes/tvheadend/sync/EpgSyncService.java
+++ b/app/src/main/java/ie/macinnes/tvheadend/sync/EpgSyncService.java
@@ -30,6 +30,7 @@ import java.util.List;
 import ie.macinnes.htsp.Connection;
 import ie.macinnes.htsp.ConnectionListener;
 import ie.macinnes.htsp.tasks.GetFileTask;
+import ie.macinnes.tvheadend.BuildConfig;
 import ie.macinnes.tvheadend.Constants;
 import ie.macinnes.tvheadend.MiscUtils;
 import ie.macinnes.tvheadend.account.AccountUtils;
@@ -135,12 +136,10 @@ public class EpgSyncService extends Service {
         final String username = mAccount.name;
         final String password = mAccountManager.getPassword(mAccount);
 
-        final String versionName = MiscUtils.getAppVersionName(mContext);
-
         // 20971520 = 20MB
         // 10485760 = 10MB
         // 1048576  = 1MB
-        mConnection = new Connection(hostname, port, username, password, "android-tvheadend (epg)", versionName, 100000);
+        mConnection = new Connection(hostname, port, username, password, "android-tvheadend (epg)", BuildConfig.VERSION_NAME, 100000);
 
         ConnectionListener connectionListener = new ConnectionListener() {
             @Override

